docs/source/en/model_doc/nllb.md

```diff
@@ -13,202 +13,121 @@ specific language governing permissions and limitations under the License.
-
-Next, [install](https://github.com/Dao-AILab/flash-attention#installation-and-features) the latest version of Flash Attention 2:
+```python
+from transformers import AutoModelForSeq2SeqLM, AutoTokenizer, BitsAndBytesConfig
 
-```bash
-pip install -U flash-attn --no-build-isolation
+bnb_config = BitsAndBytesConfig(load_in_8bit=True)
+model = AutoModelForSeq2SeqLM.from_pretrained("facebook/nllb-200-distilled-600M", quantization_config=bnb_config)
+tokenizer = AutoTokenizer.from_pretrained("facebook/nllb-200-distilled-600M")
 ```
 
-### Usage
-
-To load a model using Flash Attention 2, we can pass the argument `attn_implementation="flash_attention_2"` to [`.from_pretrained`](https://huggingface.co/docs/transformers/main/en/main_classes/model#transformers.PreTrainedModel.from_pretrained). You can use either `torch.float16` or `torch.bfloat16` precision.
+Use the [AttentionMaskVisualizer](https://github.com/huggingface/transformers/blob/main/src/transformers/utils/attention_visualizer.py#L139) to better understand what tokens the model can and cannot attend to.
 
 ```python
->>> import torch
->>> from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
-
->>> model = AutoModelForSeq2SeqLM.from_pretrained("facebook/nllb-200-distilled-600M", torch_dtype=torch.float16, attn_implementation="flash_attention_2").to("cuda").eval()
->>> tokenizer = AutoTokenizer.from_pretrained("facebook/nllb-200-distilled-600M")
-
->>> article = "Şeful ONU spune că nu există o soluţie militară în Siria"
->>> inputs = tokenizer(article, return_tensors="pt").to("cuda")
+from transformers.utils.attention_visualizer import AttentionMaskVisualizer
 
->>> translated_tokens = model.generate(
-...     **inputs, forced_bos_token_id=tokenizer.convert_tokens_to_ids("deu_Latn"), max_length=30
-... )
->>> tokenizer.batch_decode(translated_tokens, skip_special_tokens=True)[0]
-"UN-Chef sagt, es gibt keine militärische Lösung in Syrien"
+visualizer = AttentionMaskVisualizer("facebook/nllb-200-distilled-600M")
+visualizer("UN Chief says there is no military solution in Syria")
 ```
 
-### Expected speedups
-
-Below is an expected speedup diagram that compares pure inference time between the native implementation and the Flash Attention 2.
-
-<div style="text-align: center">
-<img src="https://huggingface.co/datasets/visheratin/documentation-images/resolve/main/nllb-speedup.webp">
-</div>
+## Notes
+
+- **Tokenizer behavior update (April 2023)**  
+  The tokenizer now prefixes the **source sequence** with the source language code instead of appending the target language code at the end.  
+  To restore the legacy behavior:  
+  ```python
+  from transformers import NllbTokenizer
+  tokenizer = NllbTokenizer.from_pretrained(
+      "facebook/nllb-200-distilled-600M", legacy_behaviour=True
+  )
+  ```
```

Comment:
```suggestion
- The tokenizer was updated in April 2023 to prefix the source sequence with the source language rather than the target language. This prioritizes zero-shot performance at a minor cost to supervised performance.

   <add new behavior example here>
   
   To revert to the legacy behavior, use the code example below.
   
   <add legacy_behavior example here>
   
 - For non-English languages, specify the language's [BCP-47](https://github.com/facebookresearch/flores/blob/main/flores200/README.md#languages-in-flores-200) code with the `src_lang` keyword as shown below.
 
   <add code example from https://huggingface.co/docs/transformers/v4.53.3/en/model_doc/nllb#generating-from-any-other-language-than-english>
```